docs/source/deployment/amazon_emr_serverless.md

```diff
@@ -0,0 +1,242 @@
+# Amazon EMR Serverless
+
+[Amazon EMR Serverless](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/emr-serverless.html)
+can be used to manage and execute distributed computing workloads using Apache Spark. With its serverless architecture,
+it eliminates the need to provision or manage clusters to execute your Spark jobs. Instead, EMR Serverless
+independently allocates the resources needed for each job and releases them once the job has been completed.
+
+EMR Serverless is typically used for pipelines that are either fully or partially dependent on PySpark.
+For other parts of the pipeline such as modeling, where a non-distributed computing approach may be suitable, EMR Serverless might not be needed.
+
+
+## Context
+Python applications on [Amazon Elastic MapReduce (EMR)](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-what-is-emr.html)
+have traditionally managed dependencies using [bootstrap actions](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-plan-bootstrap.html).
+This can sometimes lead to complications due to script complexity, longer bootstrapping times, and potential inconsistencies
+across cluster instances.
+
+Some applications may have requirements that necessitate a different approach, such as:
+
+- **Custom Python version:** By default, the latest EMR releases (6.10.0, 6.11.0) run Python 3.7, but an application may require Python 3.9 or later.
+- **Python dependencies:** Some applications use a range of third-party dependencies that need to be installed on both the driver and worker nodes.
+
+Even though the official AWS documentation provides methods for configuring a [custom Python version](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/using-python.html)
+and for [using custom dependencies](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/using-python-libraries.html), there are some limitations. The methods are prone to errors, can be difficult to debug, and do not ensure full compatibility (see the [FAQ section below](#faq) for more details).
+
+EMR Serverless offers a solution to the limitations described, starting with Amazon EMR 6.9.0, which supports [custom images](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/application-custom-image.html). With a custom Docker image you can package a specific Python version along
+with all required dependencies into a single immutable container. This not only ensures a consistent runtime environment
+across the entire setup but also enables control over the runtime environment.
+
+This approach helps to avoid job failures due to misconfigurations or potential conflicts with system-level packages,
+and ensures the package is portable so it can be debugged and tested locally ([see more details on validation](#optional-validate-the-custom-image)). Using the approach can achieve a repeatable, reliable environment and improve operational flexibility.
```

Comment:
```suggestion
and ensures the package is portable so it can be debugged and tested locally ([see more details on validation](#optional-validate-the-custom-image)). Using the approach can provide a repeatable, reliable environment and improve operational flexibility.
```